src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java

```diff
@@ -1254,6 +1270,112 @@ public Map<String, ObservationUnitRow> getObservationUnitsAsMap(
 			observationUnitIds);
 	}
 
+	@Override
+	public void importEnvironmentVariableValues(final Integer studyId, final Integer datasetId, final Table<String, String, String> table) {
+		final List<MeasurementVariable> measurementVariableList =
+				this.getDatasetMeasurementVariablesByVariableType(datasetId,
+						Arrays.asList(VariableType.ENVIRONMENT_DETAIL.getId(), VariableType.ENVIRONMENT_CONDITION.getId()));
+
+		if (!measurementVariableList.isEmpty()) {
+			final Map<String, MeasurementVariable> mappedVariables = new HashMap<>();
```

Comment:
Extract this to a private method e.g. createMappedVariables(measurementVariableList)